src/algebra/continued_fractions/computation/approximations.lean

```diff
@@ -0,0 +1,275 @@
+  have : ∃ (z : ℤ), gp.b = (z : K), from (of_part_num_eq_one_and_part_denom_corr_int nth_s_eq).right,
+  rwa gp_b_eq_b_n at this
+end
+
+/-
+Our next goal is to show that `Bₙ₊₁ ≥ bₙ * Bₙ`. For this, we first show that the partial denominators
+`Bₙ` grow quicker than the fibonacci sequence `nat.fib`.
+-/
+
+-- open `nat` as we will make use of fibonacci numbers.
+open nat
+
+lemma of_fib_le_continuants_aux_b : (n ≤ 1 ∨ ¬(gcf.of v).terminated_at (n - 2)) →
+  (fib n : K) ≤ ((gcf.of v).continuants_aux n).b :=
+nat.strong_induction_on n
+begin
+  clear n,
+  assume n IH hyp,
+  rcases n with _|_|n,
+  { simp [fib_succ_succ, continuants_aux] }, -- case n = 0
+  { simp [fib_succ_succ, continuants_aux] }, -- case n = 1
+  { let g := gcf.of v,  -- case 2 ≤ n
+    have : ¬(n + 2 ≤ 1), by linarith,
+    have not_terminated_at_n : ¬g.terminated_at n, from or.resolve_left hyp this,
+    obtain ⟨gp, s_ppred_nth_eq⟩ : ∃ gp, g.s.nth n = some gp, from
+      with_one.ne_one_iff_exists.elim_left not_terminated_at_n,
+    set pconts := g.continuants_aux (n + 1) with pconts_eq,
+    set ppconts := g.continuants_aux n with ppconts_eq,
+    -- use the recurrence of continuants_aux
+    suffices : (fib n : K) + fib (n + 1) ≤ gp.a * ppconts.b + gp.b * pconts.b, by
+      simpa [fib_succ_succ, add_comm,
+        (continuants_aux_recurrence s_ppred_nth_eq ppconts_eq pconts_eq)],
+    -- make use of the fact that gp.a = 1
+    suffices : (fib n : K) + fib (n + 1) ≤ ppconts.b + gp.b * pconts.b, by
+      simpa [(of_part_num_eq_one $ part_num_eq_s_a s_ppred_nth_eq)],
+    have not_terminated_at_pred_n : ¬g.terminated_at (n - 1), from
+      mt (terminated_stable $ nat.sub_le n 1) not_terminated_at_n,
+    have not_terminated_at_ppred_n : ¬terminated_at g (n - 2), from
+      mt (terminated_stable (n - 1).pred_le) not_terminated_at_pred_n,
+    -- use the IH to get the inequalities for `pconts` and `ppconts`
+    have : (fib (n + 1) : K) ≤ pconts.b, from
+      IH _ (nat.lt.base $ n + 1) (or.inr not_terminated_at_pred_n),
+    have ppred_nth_fib_le_ppconts_B : (fib n : K) ≤ ppconts.b, from
+      IH n (lt_trans (nat.lt.base n) $ nat.lt.base $ n + 1)
+      (or.inr not_terminated_at_ppred_n),
+    suffices : (fib (n + 1) : K) ≤ gp.b * pconts.b,
+      solve_by_elim [(add_le_add ppred_nth_fib_le_ppconts_B)],
+    -- finally use the fact that 1 ≤ gp.b to solve the goal
+    suffices : 1 * (fib (n + 1) : K) ≤ gp.b * pconts.b, by rwa [one_mul] at this,
+    have one_le_gp_b : (1 : K) ≤ gp.b, from
+      of_one_le_nth_part_denom (part_denom_eq_s_b s_ppred_nth_eq),
+    have : (0 : K) ≤ fib (n + 1), by exact_mod_cast (fib (n + 1)).zero_le,
+    have : (0 : K) ≤ gp.b, from le_trans zero_le_one one_le_gp_b,
+    mono }
+end
+
+/-- Shows that the `n`th denominator is greater than or equal to the `n + 1`th fibonacci number,
+that is `nat.fib (n + 1) ≤ Bₙ`. -/
+lemma of_succ_nth_fib_le_nth_denom (hyp: n = 0 ∨ ¬(gcf.of v).terminated_at (n - 1)) :
```

Comment:
Maybe state after this lemma the consequence that you want to use below, i.e., `0 \le (gcf.of v).denominators n`, as another lemma. It wouldn't need the assumption `hyp`, right?

Reply:
Added it.